packages/backend-common/src/config.ts

```diff
@@ -18,8 +18,40 @@ import { resolve as resolvePath } from 'path';
 import parseArgs from 'minimist';
 import { Logger } from 'winston';
 import { findPaths } from '@backstage/cli-common';
-import { Config, ConfigReader, JsonValue } from '@backstage/config';
-import { loadConfig } from '@backstage/config-loader';
+import { loadConfigSchema, loadConfig } from '@backstage/config-loader';
+import { AppConfig, Config, ConfigReader, JsonValue } from '@backstage/config';
+
+import { setRedactionMap } from './logging';
+
+// Fetch the schema and get all the secrets to pass to the rootLogger for redaction
+const updateRedactionMap = async (configs: AppConfig[], logger: Logger) => {
+  // Consider all packages in the monorepo when loading in config
+  const { Project } = require('@lerna/project');
+  const project = new Project();
+  const packages = await project.getPackages();
+  const localPackageNames = packages.map((p: any) => p.name);
+
+  const schema = await loadConfigSchema({ dependencies: localPackageNames });
+  const secretAppConfigs = schema.process(configs, { visibility: ['secret'] });
+  const secretConfig = ConfigReader.fromConfigs(secretAppConfigs);
+  const configMap = secretConfig.getMap();
```

Comment:
You can use `config.get()` instead, and then traverse that to flatten out all the values. Quick trick to do that can be `JSON.parse(JSON.stringify(data), (_, v) => typeof v === 'string' && values.push(v))`

Reply:
TIL About the `reviver` param 👍

Reply:
Hmm with this approach it makes it harder to get the full path to substitute the redacted secret in the logs. For example it would replace the secret with `This is the redacted Github token {{integrations.github[0].token}}`, because the `k` in the reviver fn is just the current key, not the path.

Reply:
I've changed this to ignore the key, let me know if you think I should try and get it back in again.